x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Eval.java

```diff
@@ -35,7 +36,11 @@ public List<Alias> fields() {
 
     @Override
     public List<Attribute> output() {
-        return mergeOutputAttributes(fields, child().output());
+        if (lazyOutput == null) {
+            lazyOutput = mergeOutputAttributes(fields, child().output());
+        }
```

Comment:
nit: I think it'd be neat to add a comment here that explains why we initialize this lazily. Would it be super wasteful to compute that already in the constructor instead?